jni/src/commons.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+#ifndef OPENSEARCH_KNN_COMMONS_H
+#define OPENSEARCH_KNN_COMMONS_H
+#include <jni.h>
+
+#include <vector>
+
+#include "jni_util.h"
+#include "commons.h"
+
+jlong knn_jni::commons::storeVectorData(knn_jni::JNIUtilInterface *jniUtil, JNIEnv *env, jlong memoryAddressJ,
+                                        jobjectArray dataJ, jlong initialCapacityJ, jlong startingIndexJ) {
+    std::vector<float> *vect;
+    if ((long) memoryAddressJ == 0) {
+        vect = new std::vector<float>((long)initialCapacityJ);
+    } else {
+        vect = reinterpret_cast<std::vector<float>*>(memoryAddressJ);
+    }
+    long startingIndex = (long) startingIndexJ;
+    int dim = jniUtil->GetInnerDimensionOf2dJavaFloatArray(env, dataJ);
+    auto dataset = jniUtil->Convert2dJavaObjectArrayToCppFloatVector(env, dataJ, dim);
```

Comment:
Should we try to directly store this into vect? Im thinking maybe we can pass the vector by ref into this function instead of creating `floatVectorCpp` and then copying: https://github.com/opensearch-project/k-NN/blob/main/jni/src/jni_util.cpp#L224-L257

Reply:
Yeah I think that will be better. This can further cut down computations. Let me make the change